lib/sequelize.js

```diff
@@ -893,9 +893,12 @@ module.exports = (function() {
         }
       });
 
+      // Iterate daos and sync.  Collect results into array
       return Promise.reduce(daos, function(total, dao) {
-        return dao.sync(options);
-      }, null);
+        return total.push(dao.sync(options));
```

Comment:
`push` returns the target so the next iteration of reduce will fail.
You need to use something like:

```js
return dao.sync(options).then(function (result) {
  return total.concat([result]);
});
```